### PR TITLE
created basic Statistic and Action

### LINKED ISSUE #10: Characters during combat: MVP
- implement `weasel::CharacterRules`

## Acceptance criteria
- Creatures must have the following stats:
  - hit points
  - all existing abilities from the compendium
  - proficiency bonus
  - all existing skills from the compendium
- Creatures must have the following abilities:
  - move
  - attack

### PR DESCRIPTION
**Pull request by Trisfald**
Very very basic `struct`s to represent a character set of Statistics and Actions.

closes #10

### CODE REVIEW

File: src/rules/core/action.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    id: ActionId,
+    value: ActionValue,
+}
+
+impl Action {
+    fn new(id: ActionId, value: ActionValue) -> Self {
+        Self { id, value }
+    }
+
+    /// Returns a reference to `()` if this action is of the correct type, otherwise `None`.
```
Comment (LesnyRumcajs) [inline src/rules/core/action.rs:20]: Seems to me like `Result` would be better suited for this one

File: src/rules/core/action.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    pub fn movement(&self) -> Option<()> {
+        if let ActionValue::Movement = &self.value {
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    /// Returns a reference to `()` if this action is of the correct type, otherwise `None`.
+    pub fn attack(&self) -> Option<()> {
```
Comment (LesnyRumcajs) [inline src/rules/core/action.rs:30]: same

File: src/rules/core/statistic.rs [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+impl StatisticValue {
+    fn apply_change(&mut self, change: &StatisticChange) {
+        let mut successful = false;
+        use StatisticValue::*;
+        match change {
+            StatisticChange::HitPoints(change) => {
+                if let HitPoints(value) = self {
+                    if *change > 0 {
+                        value.add(change.abs() as u16);
```
Comment (LesnyRumcajs) [inline src/rules/core/statistic.rs:124]: the r-value here could be moved to a variable to make it more DRY

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: src/rules/core/action.rs (hunk 1 above)
Comment (Trisfald) [reply src/rules/core/action.rs]: done!

File: src/rules/core/action.rs (hunk 2 above)
Comment (Trisfald) [reply src/rules/core/action.rs]: done!

(I will do the same for statistics in another PR)

File: src/rules/core/statistic.rs (hunk 3 above)
Comment (Trisfald) [reply src/rules/core/statistic.rs]: done

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**